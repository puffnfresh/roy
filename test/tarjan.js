var stronglyConnectedComponents = require('../src/tarjan.js').stronglyConnectedComponents;
var _ = require('underscore');

var a = {id: "a"};
var b = {id: "b"};
var c = {id: "c"};
var d = {id: "d"};
var e = {id: "e"};
var f = {id: "f"};
var g = {id: "g"};
var h = {id: "h"};
var vertices = [a, b, c, d, e, f, g, h];
var edges = {
    "a": [b],
    "b": [c, e, f],
    "c": [d, g],
    "d": [c, h],
    "e": [a, f],
    "f": [g],
    "g": [f],
    "h": [d, g]
};

var components = stronglyConnectedComponents({vertices: vertices, edges: edges});

_.each(components, function(component) {
           component = _.sortBy(component, function(vertex) {
                                    return vertex.id;
                                });

           _.each(component, function(vertex) {
                      console.log(vertex.id);
                  });
           console.log();
       });