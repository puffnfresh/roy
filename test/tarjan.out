f
g

c
d
h

a
b
e

